src/Compilers/Core/Desktop/EncodedStringText.cs

```diff
@@ -37,10 +37,11 @@ internal static class EncodedStringText
         /// <paramref name="defaultEncoding"/> is null and the stream appears to be a binary file.
         /// </exception>
         /// <exception cref="IOException">An IO error occurred while reading from the stream.</exception>
-        internal static SourceText Create(Stream stream, Encoding defaultEncoding = null, SourceHashAlgorithm checksumAlgorithm = SourceHashAlgorithm.Sha1)
+        internal static SourceText Create(Stream stream, Encoding defaultEncoding, SourceHashAlgorithm checksumAlgorithm = SourceHashAlgorithm.Sha1)
         {
             Debug.Assert(stream != null);
             Debug.Assert(stream.CanRead && stream.CanSeek);
+            Debug.Assert(defaultEncoding != null);
```

Comment:
Paul is correct. I missed that. We might need additional bool detectEncoing parameter